jdaviz/configs/cubeviz/plugins/spectral_extraction/spectral_extraction.py

```diff
@@ -315,28 +315,48 @@ def uncert_cube(self):
     def spectral_display_unit(self):
         return astropy.units.Unit(self.app._get_display_unit(self.slice_display_unit_name))
 
+    @property
+    def mask_non_science(self):
+        # Aperture masks begin by finding removing from consideration any pixel
+        # set to NaN, which corresponds to a pixel on the "non-science" portions
+        # of the detector. For JWST spectral cubes, these pixels are marked in
+        # the DQ array with flag `513`.
+        return np.logical_not(
+            np.isnan(self.dataset.selected_obj.flux.value)
+        ).astype(float)
+
     @property
     def aperture_weight_mask(self):
         # Exact slice mask of cone or cylindrical aperture through the cube. `weight_mask` is
         # a 3D array with fractions of each pixel within an aperture at each
         # wavelength, on the range [0, 1].
         if self.aperture.selected == self.aperture.default_text:
             # Entire Cube
-            return np.ones_like(self.dataset.selected_obj.flux.value)
-        return self.aperture.get_mask(self.dataset.selected_obj,
-                                      self.aperture_method_selected,
-                                      self.spectral_display_unit,
-                                      self.reference_spectral_value if self.wavelength_dependent else None)  # noqa
+            return self.mask_non_science
+
+        return (
+            self.mask_non_science *
+            self.aperture.get_mask(
```

Comment:
would there be any use for this logic _within_ `aperture.get_mask` instead of a property in the plugin?

Reply:
Since `ApertureSubsetSelect` is meant to be general, I'd advocate for this particular solution to be associated with spectral extraction. There are different ways that you might want to handle NaN masks, for example, in spectral cubes vs images. If you had a flat-fielded image that produced a NaN, you might want your aperture photometry to do something like zero-filling or interpolating over the NaN, in which case you would still count the sky area in that pixel.